docs/articles/user_acquisition_analytics.md

```diff
@@ -0,0 +1,439 @@
+).properties(height=400, width=1200)
+```
+
+![new users per signup date](../assets/use_cases/user_acquisition_analytics/new_users-signup_date.png)
+
+[observations] First, the distribution confirms that our second campaign (December) works much better than the first (August).
+Second, the jump in signups at 2023-12-21 is due to the second campaign (our data is exclusively campaign-related). To analyze the two campaigns, we need two periods: a first period of 65 days and a second of 185 days seems appropriate.
+Of our 8k users, roughly 2k subscribed in the first campaign period (65 days), and 6k in the second. ...It's possible that this 6k number includes subscriptions not just from new (December) campaign ad clicks, but clicks on old (August) campaign ads that occur after seeing new campaign ads. We'll see how this breaks out by... ad creatives [?]... (below)...
+
+so far.. we know that:
+
+- many more users signed up in response to the influencer-backed ad creatives (second campaign) than the first campaign
+- the vast majority of users are low activity, and more recent subscribers
+
+We don't know....
+
+- what kind of ad creatives have similar meanings and possibly outcomes as a result
+- what kinds of users (in terms of activity level) responded to what kind of ad_creatives (in terms of semantic meaning)
+
+We can use embedding and Superlinked spaces to help reveal answers to our remaining questions..
+
+
+## Embedding with Superlinked
+
+Now, let's use Superlinked to embed our data in a semantic space - to:
+ 
+1. inform the model re *which ad creatives* generated **signups**, and *which users* signed up... 
+2. group ad creatives that have similar meanings..
+
+Define a schema for our user data:
+
+```python
+class UserSchema(Schema):
+    ad_creative: String
+    activity: Float
+    signup_date: Timestamp
+    id: IdField
+```
+
+```python
+user = UserSchema()
+```
+
+Now we create a semantic space for our ad_creatives using a text similarity model. Then encode user activity into a numerical space to represent users' activity level. We also encode the signup date into a recency space, allowing our clustering algorithm to take account of the two specific periods of signup activity (following our two campaign start dates).
+
+```python
+creative_space = TextSimilaritySpace(
+    text=user.ad_creative, model="sentence-transformers/all-mpnet-base-v2"
+)
+
+activity_space = NumberSpace(
+    number=user.activity, mode=Mode.SIMILAR, min_value=0.0, max_value=1.0
+)
+
+recency_space = RecencySpace(
+    timestamp=user.signup_date,
+    period_time_list=[PeriodTime(timedelta(days=65)), PeriodTime(timedelta(days=185))],
+    negative_filter=0.0,
+)
+```
+
+Now, let's plot our recency scores by date.
+
+```python
+recency_plotter = RecencyPlotter(recency_space, context_data=EXECUTOR_DATA)
+recency_plotter.plot_recency_curve()
+```
+
+![](../assets/use_cases/user_acquisition_analytics/recency_scores-by-date.png)
```

Comment:
@morkapronczay 
How exactly do the recency scores work in this case?

Reply:
Not sure what kind of information are you looking for here. Recency scores are configured in the line where `RecencySpace` is created, and the NOW of the application is configured in the beginning (so the notebook runs reproducibly everytime).